frontend/src/pages/users/UserAdd.tsx

```diff
@@ -0,0 +1,275 @@
+import { FormEvent, useEffect, useRef, useState } from "react";
+import { logger } from "../../util/Logger";
+import { Toast } from "primereact/toast";
+import BackButton from "../../components/buttons/BackButton";
+import { useNavigate, useParams } from "react-router-dom";
+import { InputText } from "primereact/inputtext";
+import { Password } from "primereact/password";
+import { InputSwitch } from "primereact/inputswitch";
+import {
+  passwordValueCheck,
+  pwCheckFooter,
+  pwCheckHeader,
+} from "../../util/PWValueCheck";
+import { showFailure, showSuccess } from "../../components/Toast";
+import ConfirmPopup from "../../components/popups/ConfirmPopup";
+import DeletePopup from "../../components/popups/DeletePopup";
+import DeleteButton from "../../components/buttons/DeleteButton";
+import { USER_API } from "../../apis/users/UserAPI";
+
+export default function VendorAdd() {
+  // From URL
+  const { id } = useParams();
+  const isUserAddPage = id === undefined;
+  const [isModifiable] = useState<boolean>(true);
+
+  const [userName, setUserName] = useState<string>("");
+  const [password1, setPassword1] = useState<string>("");
+  const [password2, setPassword2] = useState<string>("");
+  const [isAdmin, setIsAdmin] = useState<boolean>(false);
+
+  const [isConfirmationPopupVisible, setIsConfirmationPopupVisible] =
+    useState<boolean>(false);
+  const [deletePopupVisible, setDeletePopupVisible] = useState<boolean>(false); // Whether the delete popup is visible
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [isPageDeleteable, setIsPageDeleteable] = useState<boolean>(true);
+
+  // Toast is used for showing success/error messages
+  const toast = useRef<Toast>(null);
+
+  // Load the Genre data on page load
+  useEffect(() => {
+    USER_API.getUserDetail({ id: id! })
+      .then((response) => {
+        setUserName(response.username);
+        setIsAdmin(response.is_staff);
+      })
+      .catch(() => showFailure(toast, "Could not fetch user data"));
+  }, []);
+
+  // Called to make delete pop up show
+  const deleteUserPopup = () => {
+    logger.debug("Delete Purchase Order Clicked");
+    setDeletePopupVisible(true);
+  };
+
+  // Call to actually delete the element
+  const deleteUserFinal = () => {
+    logger.debug("Edit Purchase Order Finalized");
+    setDeletePopupVisible(false);
+    USER_API.deleteUser({
+      id: id!,
+    })
+      .then(() => {
+        showSuccess(toast, "User Deleted");
+        navigate("/users");
+      })
+      .catch(() => showFailure(toast, "User Failed to Delete"));
+  };
+
+  const onSubmit = (): void => {
+    logger.debug(
+      "Add User Submitted:",
+      userName,
+      password1,
+      password2,
+      isAdmin
+    );
+
+    if (!isUserAddPage) {
```

Comment:
This triply nested if statement is hard on the eyes. I think it can be broken up into multiple smaller functions.

The outermost one could be
if(!isUserAddPage) {
   tryToModifyUser
} else {
  tryToAddUser()
}

I'm actually not sure what the second one does inside the !isUserAddPage, but it can be broken down into more easily understandable logic too. Also the same modifyUser call can be extracted out